### PR TITLE
py: Escape quotes are recognised correctly.

### PR DESCRIPTION
**Pull request by hosaka**
Fix for the mentioned #1419 plus the test coverage. I am not too sure about the return on line 87, for example a call like:

``` python
print(\")
```

should be treated as a syntax error. Perhaps someone can comment?

### CODE REVIEW

File: py/repl.c [hunk 1]
```diff
@@ -59,12 +59,7 @@ bool mp_repl_continue_with_input(const char *input) {
         || str_startswith_word(input, "class")
         ;
 
-    // check for unmatched open bracket or quote
-    #define Q_NONE (0)
-    #define Q_1_SINGLE (1)
-    #define Q_1_DOUBLE (2)
-    #define Q_3_SINGLE (3)
-    #define Q_3_DOUBLE (4)
```
Comment (dpgeorge) [inline py/repl.c]: You should keep these defines in the .c file.  There's no reason to polute the namespace of someone importing repl.h.

**Comment by dpgeorge** [comment]
Thanks!

Yes, `print(\")` is difficult to catch.  CPython works by actually parsing each line as it comes in.  We don't and won't do that.  Maybe just ignore this corner case?

File: py/repl.c (hunk 1 above)
Comment (alex-march) [inline py/repl.c]: Will put it back! The motivation behind it: they were getting defined every time the mp_repl_continue_with_input was called. Perhaps just put it just below the MICROPY_HELPER_REPL on line 33?

**Comment by alex-march** [comment]
@dpgeorge Okay, if that's the case I will remove the else branch.

File: py/repl.c (hunk 1 above)
Comment (dpgeorge) [inline py/repl.c]: It's a macro that's substituted at compile time.  Only done once.

Comment (alex-march) [inline py/repl.c]: Apologies, indeed, that's very silly of me.

**Comment by dpgeorge** [comment]
Merged, thank you!

**Pull request closed without merging**